opm/grid/cpgrid/processEclipseFormat.cpp

```diff
@@ -200,15 +209,106 @@ namespace cpgrid
+                        bottom_cell_info = ecl_grid.getCellAndBottomCenterNormal(cell_bottom);
+                        const auto& [ cell_center_top, face_center, area_normal ] = top_cell_info;
+                        const auto& cell_center_bottom = std::get<0>(bottom_cell_info);
+                        cells_between.push_back(cell_top);
+
+                        auto compute_half_trans =
+                            [&face_center, &area_normal](const std::array<double,3>& cell_center,
+                                                         double perm)
+                            {
+                                auto half_trans = perm;
+                                std::array<double, 3> distance;
+                                std::transform(cell_center.begin(), cell_center.end(), face_center.begin(),
+                                               distance.begin(), std::minus<double>());
+                                half_trans *= std::abs(std::inner_product(area_normal.begin(), area_normal.end(), distance.begin(), 0.));
+                                half_trans /= std::inner_product(distance.begin(), distance.end(), distance.begin(), 0.);
+                                return half_trans;
+                            };
+
+                        auto half_trans_top = compute_half_trans(cell_center_top,
+                                                                 permZ[cell_top]);
+                        auto half_trans_bottom = compute_half_trans(cell_center_bottom,
+                                                                    permZ[cell_bottom]);
+
+                        if (std::abs(half_trans_top) < 1e-30 || std::abs(half_trans_bottom) < 1e-30)
+                            *trans = 0.0;
+                        else
+                            *trans = 1.0 / (1.0/half_trans_top + 1.0/half_trans_bottom);
+
+                        ++trans;
+                    }
 
-        // Add explicit NNCs.
-        if (ecl_state) {
+                    // Possibly overwrite with specified TRANZ values.
+                    if (fp.tran_active("TRANZ"))
+                    {
+                        fp.apply_tranz_global(cells_between, trans_between);
+                    }
+
+                    // Apply Multipliers
+
+                    // \todo FIXME We assume here that MULTZ does not change in the SCHEDULE or such changes have
+                    //       no effect here. This might be wrong and need fixing. Which basically means we need to store
+                    //       all the intermediate transmissibilities for the harmonic average and later apply additional
+                    //       multipliers
+                    const auto& transMult = ecl_state->getTransMult();
+                    trans = trans_between.begin();
+
+                    for (std::size_t cell_top = cell1, cell_bottom = cell_top + ecl_grid.getNX() * ecl_grid.getNY();
```

Comment:
In this loop, trans is incremented in the loop body, unlike the one above. Is there a reason for the difference, or could it be unified?

Reply:
Changed the loop above to also increment in the for statement. Seemed safer.